### PR TITLE
Add option to log messages to terminal

### PR DESCRIPTION
**Pull request by Schobs**
Fixes #{issue_number}.

### Description
Adds option to log to terminal. THis is useful since we want to avoid print statements, and in interactive examples it is important to print to terminal. This will not affect any existing code since it defaults to False.

### Status
**Ready**

If you've been given access to pykale with role Triage or above, on the right (delete these after selection):

- Select **one most appropriate** label.
- **When** your pull request is **ready** for review, select a reviewer. Use the suggested one if unsure.

If you cannot see an option to select a reviewer/label, that means one maintainer will get notified upon your pull request and then select for you.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

**Review by Mdnaimulislam — APPROVED** [review]

**Review by Mdnaimulislam — COMMENTED** [review]
@Schobs Didn't pass all the checks.

**Review by Mdnaimulislam — APPROVED** [review]
All okay.

File: kale/utils/logger.py [hunk 1]
```diff
@@ -38,4 +40,10 @@ def construct_logger(name, save_dir):
     gitdiff_patch = os.path.join(save_dir, file_no_ext + ".gitdiff.patch")
     os.system(f"git diff HEAD > {gitdiff_patch}")
 
+    if log_to_terminal:
```
Comment (haipinglu) [inline kale/utils/logger.py:43]: @Schobs  Update https://github.com/pykale/pykale/blob/main/tests/utils/test_logger.py to test this option to improve the coverage.

**Review by haipinglu — CHANGES_REQUESTED** [review]
@Schobs Some minor changes needed before merging.

**Review by Mdnaimulislam — APPROVED** [review]
test_logger is coveraging 100% of logger. LGTM. Well done.

**Review by haipinglu — APPROVED** [review]

**Pull request merged**